apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -76,13 +80,39 @@ type APIManagerStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	Conditions []APIManagerCondition `json:"conditions,omitempty" protobuf:"bytes,4,rep,name=conditions"`
+	// Current state of the APIManager resource.
+	// Conditions represent the latest available observations of an object's state
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions common.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,2,rep,name=conditions"`
 
 	// APIManager Deployment Configs
 	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Deployments",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podStatuses"
 	Deployments olm.DeploymentStatus `json:"deployments"`
 }
 
+func (s *APIManagerStatus) Equals(other *APIManagerStatus, logger logr.Logger) bool {
+	// Marshalling sorts by condition type
+	currentMarshaledJSON, _ := s.Conditions.MarshalJSON()
+	otherMarshaledJSON, _ := other.Conditions.MarshalJSON()
+	if string(currentMarshaledJSON) != string(otherMarshaledJSON) {
+		diff := cmp.Diff(string(currentMarshaledJSON), string(otherMarshaledJSON))
+		logger.V(1).Info("Conditions not equal", "difference", diff)
+		return false
+	}
+
+	// TODO should we sort the DCs? when we set the status section we are
```

Comment:
See this TODO

Reply:
I've replaced the TODO by a comment indicating that at that point the DCs should be  sorted already